.github/CODESTYLE.md

```diff
@@ -0,0 +1,305 @@
+
+    Example:
+```fortran
+subroutine feed_pets(cats, dogs, food, angry)
+    !! Feeds your cats and dogs, if enough food is available. If not enough
+    !! food is available, some of your pets will get angry.
+
+    ! Arguments
+    integer, intent(in)  :: cats
+    !! The number of cats to keep track of.
+    integer, intent(in)  :: dogs
+    !! The number of dogs to keep track of.
+    real, intent(inout)  :: food
+    !! The amount of pet food (in kilograms) which you have on hand.
+    integer, intent(out) :: angry
+    !! The number of pets angry because they weren't fed.
+
+    !...
+    return
+end subroutine feed_pets
+```
+
+* Avoid “side effects” (return values are the only thing modified: files are not created, module/global variables are not changed, etc). Use pure functions?
+* `implicit none` for all program units
+* Use `use, only:` as much as possible.
+* Restrict number of passed variables per line in the function/subroutine definition/call to 4 max. Line up variables vertically for ease of reading and counting. Match the call layout to the definition layout.
+* Have a local variables section separate from passed variables
+
+```fortran
+# WRONG
+some_function(a,b,c,d,e,f,g,h &
+,i,j,k,l)
+real :: a,b,c,d,e,f
+Ingeter :: localVar
+Real, intent(in) :: h,i,j,k,l
+
+call some_function(a,b,c,d &
+,e,f &
+g,h,i,j,k,l)
+
+# CORRECT
+some_function(a, b, c, d &
+             ,e, f, g, h &
+             ,i, j, k, l )
+Use module_great_stuff: only, shiny_object, rad_tool
+Implicit none
+real, intent(in)  :: a !! acceleration   (m/s2)
+real, intent(in)  :: b !! brownian coeff (-)
+real, intent(out) :: c !! celerity       (m/s)
+… (etc) …
+real, intent(in)  :: l !! lousy variable (kgmsK)
+
+!! Local variables
+integer :: localVar
+
+call some_function(a, b, c, d &
+                  ,e, f, g, h &
+                  ,i, j, k, l )
+```
+
+# Variable Scoping and Definition
+* Minimize global data.
+* Global data should be limited to constants as much as possible.
+* Define and use global type parameters.
+* Make all derived type components private.
+
+# Run-time error messages
+* Must indicate the function/routine in which they occur.
+* Should be informative.
+
+# Compiler warning messages
+* Compilers generally issue two types of messages: warnings and errors. Extent of compiler warning messages can typically be tuned with a compiler option. Compiler warnings normally do not stop the compile process, but can still result in run-time problems. Compiler errors do stop the compile process, forcing the developer to fix the problem and recompile.
+STANDARD:
+* When available, compiler options should be set to produce the maximum number of warning messages.
+* Compiler and linker warnings shall be treated as errors and fixed.
+
+# Other general guidelines
```

Comment:
Maybe consider putting some of this up top?

Reply:
This is basically just a literal translation of the standards written by the fortran devs here. I do not write fortran currently so I will leave this to their discretion. @jmccreight